app/src/main/kotlin/com/wire/android/ui/userprofile/UserProfileViewModel.kt

```diff
@@ -31,32 +29,69 @@ import kotlinx.coroutines.withContext
 class UserProfileViewModel @Inject constructor(
     private val navigationManager: NavigationManager,
     private val dataStore: UserDataStore,
-    private val uploadUserAvatar: UploadUserAvatarUseCase,
+    private val getPublicAssetUseCase: GetPublicAssetUseCase,
     private val getSelf: GetSelfUserUseCase
 ) : ViewModel() {
 
     var userProfileState by mutableStateOf(SelfUserProfileState())
         private set
 
     init {
-        // TODO: here we should have a loading state as the first initial state of the screen
         viewModelScope.launch {
-            getSelf().collect {
-                userProfileState = SelfUserProfileState(
-                    status = UserStatus.AVAILABLE,
-                    fullName = it.name.orEmpty(),
-                    userName = it.handle.orEmpty(),
-                    teamName = it.team,
-                    // Add some mocked team
-                    otherAccounts = listOf(
-                        OtherAccount("someId", "", it.name.orEmpty(), "Wire Swiss GmbH"),
-                        OtherAccount("someId", "", "B. A. Baracus", "The A-Team"),
+            fetchSelfUser()
+        }
+    }
+
+    private suspend fun fetchSelfUser() {
+        // Show progress bar on initial load
+        showLoadingAvatar(true)
+
+        viewModelScope.launch {
+            getSelf().collect { selfUser ->
+                with(selfUser) {
+                    // Load user avatar raw image data
+                    val avatarAssetRaw = completePicture?.let { getAvatarAsByteArrayOrNull(it) }
```

Comment:
we are not handeling any failures here

Reply:
If the avatar asset raw is null, then the default fallback picture will be shown

Reply:
that is true, but we are calling : getPublicAssetUseCase(avatarAssetId) inside getAvatarAsByteArrayOrNull() so there could some failure from the use case and we are not handeling but we are just displaying the default picture and not show any snackbar or anything, is it something we want to do ?

Reply:
The error here will be when downloading the avatar picture, not when trying to upload it as the design specs suggest, so not sure what should I show here, maybe some snack bar saying "There was an error while fetching the user info, check your internet connection" ? But it is also true that I didn't add the snackbar when the uploads fails so will do it right away.

Reply:
oki doki, thanks for explanation, your call on that one :P I am fine with both